zrml/court/src/migrations.rs

```diff
@@ -14,3 +14,220 @@
+
+        let res = crate::CourtPool::<T>::translate::<OldCourtPoolOf<T>, _>(|old_pool_opt| {
+            old_pool_opt.map(|mut old_pool| {
+                <CourtPoolOf<T>>::truncate_from(
+                    old_pool
+                        .iter_mut()
+                        .map(|old_pool_item: &mut OldCourtPoolItemOf<T>| CourtPoolItemOf::<T> {
+                            stake: old_pool_item.stake,
+                            court_participant: old_pool_item.court_participant.clone(),
+                            consumed_stake: old_pool_item.consumed_stake,
+                            joined_at: old_pool_item.joined_at,
+                            uneligible_index: old_pool_item
+                                .joined_at
+                                .checked_div(&T::InflationPeriod::get())
+                                // because inflation period is not zero checked_div is safe
+                                // however, if not use the current period index
+                                // 5_184_000 is the block of the inflation reward distribution
+                                // 5_184_000 / 216_000 (blocks per 30 day inflation period) = 24
+                                .unwrap_or(24u64.saturated_into::<BlockNumberOf<T>>()),
+                            // using old_pool_item.stake leads to all joins in period 24
+                            // to be uneligible, which is exactly what we want
+                            // if using zero, all joins of period 24 would be eligible,
+                            // but only joins in 23 waited the full period yet
+                            // to understand the calculation of the eligible stake look into handle_inflation
+                            uneligible_stake: old_pool_item.stake,
+                        })
+                        .collect::<Vec<_>>(),
+                )
+            })
+        });
+        match res {
+            Ok(_) => log::info!("MigrateCourtPoolItems: Success!"),
+            Err(e) => log::error!("MigrateCourtPoolItems: Error: {:?}", e),
+        }
+
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+
+        StorageVersion::new(COURT_NEXT_STORAGE_VERSION).put::<Court<T>>();
+        total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
+        log::info!("MigrateCourtPoolItems: Done!");
+        total_weight
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        log::info!("MigrateCourtPoolItems: Preparing to migrate old pool items...");
+        Ok(vec![])
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_previous_state: Vec<u8>) -> Result<(), &'static str> {
+        let court_pool = crate::CourtPool::<T>::get();
+        log::info!(
+            "MigrateCourtPoolItems: post-upgrade executed. Migrated {:?} pool items",
+            court_pool.len()
+        );
+        Ok(())
+    }
```

Comment:
We should implement this logic to verify that the migrations are successful on Battery Station and Zeitgeist chains.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/1285/commits/3b3762090538b2a5d83ba15f38161e801e905df4